sdk/src/main/java/com/horizen/account/api/rpc/types/EthereumTransactionView.java

```diff
@@ -7,68 +7,68 @@
 import com.horizen.evm.utils.Hash;
 import com.horizen.serialization.Views;
 import com.horizen.utils.BytesUtils;
-import org.web3j.utils.Numeric;
 
 import java.math.BigInteger;
-import java.util.ArrayList;
 import java.util.Objects;
 
 @JsonView(Views.Default.class)
 public class EthereumTransactionView {
     public final Hash blockHash;
-    public final String blockNumber;
+    public final BigInteger blockNumber;
     public final Address from;
     public final Hash hash;
-    public final String transactionIndex;
-    public final String type;
-    public final String nonce;
+    public final BigInteger transactionIndex;
+    public final BigInteger type;
+    public final BigInteger nonce;
     public final Address to;
-    public final String gas;
-    public final String value;
-    public final String input;
-    public final String maxPriorityFeePerGas;
-    public final String maxFeePerGas;
-    public final String gasPrice;
-    public final ArrayList<Object> accessList;
-    public final String chainId;
-    public final String v;
-    public final String r;
-    public final String s;
+    public final BigInteger gas;
+    public final BigInteger value;
+    public final byte[] input;
+    public final BigInteger maxPriorityFeePerGas;
+    public final BigInteger maxFeePerGas;
+    public final BigInteger gasPrice;
+    // as we don't support transactions with accessList we don't include it here
+    // note: accessList should be either the value used in the TX or empty-array []
+    // if the transaction type does not support it, it should not be present at all - don't serialize as "null"
+//    public final ArrayList<Object> accessList;
```

Comment:
Does it mean that we need to skip it for `legacy` tx type, but keep it as `[]` for eip-1559? because eip1559 actually supports this field (but in our case it is always empty)

Reply:
Yes, see also the comment about `null`. Legacy TXs must definitely not include `"accessList":null`, for the TX types that support it, it should be empty-array instead `[]`. It was inconsistent before (null in every TX) and still is (now it is not included at all). We should fix this.

Reply:
I think it is important to fix now.
Waht we can do:
1) return back `accessList` field
2) set to empty array for eip1559 and to null for legacy
3) use `@JsonInclude(Include.NON_NULL)` annotation for this field.

As a result I expect it to be serialized as expected in all cases.

Reply:
done